CMakeLists.txt

```diff
@@ -0,0 +1,162 @@
+
+# postw90.x specific
+set(srcp90 ${PROJECT_SOURCE_DIR}/src/postw90/berry.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/boltzwann.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/dos.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/geninterp.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/get_oper.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/gyrotropic.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/kpath.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/kslice.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/postw90_common.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/postw90_readwrite.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/postw90_types.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/spin.F90
+    ${PROJECT_SOURCE_DIR}/src/postw90/wan_ham.F90)
+
+# library specific
+set(srclib ${PROJECT_SOURCE_DIR}/src/library_interface.F90)
+
+# library objects to avoid recompilation of common objects
+add_library(obj_common OBJECT ${src})
+add_library(obj_wann90 OBJECT ${srcw90} $<TARGET_OBJECTS:obj_common>)
+add_library(obj_post90 OBJECT ${srcp90} $<TARGET_OBJECTS:obj_common>)
+add_library(obj_lib OBJECT ${srclib} $<TARGET_OBJECTS:obj_common> $<TARGET_OBJECTS:obj_wann90> )
+
+# shared library
+add_library(dynamic_lib SHARED $<TARGET_OBJECTS:obj_lib> $<TARGET_OBJECTS:obj_common> $<TARGET_OBJECTS:obj_wann90>)
+set_property(TARGET dynamic_lib PROPERTY VERSION ${PROJECT_VERSION})
+#set(soserial "2") # version 2 of library, but wannier90 version 4.0.0, to messy
+#set_property(TARGET dynamic_lib PROPERTY SOVERSION ${soserial})
+set_property(TARGET dynamic_lib PROPERTY OUTPUT_NAME wannier90)
+install(TARGETS dynamic_lib LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+# static library
+add_library(static_lib STATIC $<TARGET_OBJECTS:obj_lib> $<TARGET_OBJECTS:obj_common> $<TARGET_OBJECTS:obj_wann90>)
+set_property(TARGET static_lib PROPERTY VERSION ${PROJECT_VERSION})
+set_property(TARGET static_lib PROPERTY OUTPUT_NAME wannier90)
+install(TARGETS static_lib ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+# wannier90.x executable
+add_executable(wannier90.x ${PROJECT_SOURCE_DIR}/src/wannier_prog.F90 $<TARGET_OBJECTS:obj_common> $<TARGET_OBJECTS:obj_wann90>)
+target_link_libraries(wannier90.x dynamic_lib ${LIBS})
+install(TARGETS wannier90.x RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+# postw90.x executable
+add_executable(postw90.x ${PROJECT_SOURCE_DIR}/src/postw90/postw90.F90 $<TARGET_OBJECTS:obj_common> $<TARGET_OBJECTS:obj_post90>)
+target_link_libraries(postw90.x ${LIBS})
+install(TARGETS postw90.x RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+# include files
+install(FILES ${PROJECT_SOURCE_DIR}/src/library_interface.F90 DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/w90_helper_types.mod DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+# pkg-config
+configure_file(wannier90.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/wannier90.pc)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/wannier90.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig COMPONENT pkgconfig)
```

Comment:
Wrap all `install()` against a `WANNIER90_INSTALL` or `PROJECT_IS_TOPLEVEL`. This will make `FetchContent` incompatible otherwise